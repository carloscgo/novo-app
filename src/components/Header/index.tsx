import { Link } from "react-router-dom";
import { Container, Nav, Navbar, NavDropdown } from 'react-bootstrap';
import {
  PropsHeader
} from '../../utils/interfaces';
import { searchRoute } from '../../utils/services';
import NavHeader from './styles';
import Menu from "../Menu";

const Header = ({ brand, userName, menu }: PropsHeader) => {
  return (
    <Container fluid className="p-0">
      <NavHeader expand="lg">
        <Container>
          <Navbar.Brand className="brand">
            <Link to={searchRoute('home')}>{brand}</Link>
          </Navbar.Brand>

          <Navbar.Collapse className="menu-user" id="user-navbar-nav">
            <Nav>
              <NavDropdown title={userName}>
                <NavDropdown.Item href="#action">Action</NavDropdown.Item>
              </NavDropdown>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </NavHeader>

      <Navbar className="menu-bar" expand="lg">
        <Container className="align-items-start">
          <Menu items={menu} onClick={(e) => console.log(e)} selection={null} />
        </Container>
      </Navbar>
    </Container>
  )
}

export default Header