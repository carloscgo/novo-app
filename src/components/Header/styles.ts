import styled from 'styled-components';
import Navbar from 'react-bootstrap/Navbar';

const Header = styled(Navbar)`
  background-color: var(--main-color);
  justify-content: space-between;
  width: 100% !important;

  a:not(.dropdown-item), .dropdown.show > a {
    color: var(--text-white);

    &:hover {
      color: var(--text-white);
    }
  }

  .brand * {
    font-style: 1.5rem;
    font-weight: 700;
    text-decoration: none;
  }

  .menu-user {
    display: block !important;
    width: 200px;
    flex-grow: 0;
    justify-content: flex-end;
  }

  .menu-bar {
    background-color: var(--bg-white) !important;
  }
`

export default Header