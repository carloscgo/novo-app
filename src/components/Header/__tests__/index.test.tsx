/**
 *
 * Tests for Header
 *
 */

import Header from '../index'

import { render, BrowserRouter } from '../../../../jest.setup'

describe('<Header />', () => {
  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation(() => { })
  })

  const renderComponent = () => render(
    <BrowserRouter>
      <Header brand="Brand" userName="Carlos Camacho" menu={[]} />
    </BrowserRouter>
  )

  it('expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')

    renderComponent()

    expect(spy).not.toHaveBeenCalled()
  })
})
