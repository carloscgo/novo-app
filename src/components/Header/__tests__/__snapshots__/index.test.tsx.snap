// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> expect render with onSelect event 1`] = `
<body>
  <div>
    <div
      class="p-0 container-fluid"
    >
      <nav
        class="sc-hLBbgP ASbcf navbar navbar-expand-lg navbar-light"
      >
        <div
          class="container"
        >
          <span
            class="brand navbar-brand"
          >
            <a
              href="/dashboard"
            >
              Brand
            </a>
          </span>
          <div
            class="menu-user navbar-collapse collapse"
            id="user-navbar-nav"
          >
            <div
              class="navbar-nav"
            >
              <div
                class="nav-item dropdown"
              >
                <a
                  aria-expanded="false"
                  class="dropdown-toggle nav-link"
                  href="#"
                  role="button"
                  tabindex="0"
                >
                  Carlos Camacho
                </a>
              </div>
            </div>
          </div>
        </div>
      </nav>
      <nav
        class="menu-bar navbar navbar-expand-lg navbar-light"
      >
        <div
          class="align-items-start container"
        >
          <div
            class="sc-bcXHqe jzLKGg"
          >
            <div
              class="item-width align-items-start selected navbar-nav"
            >
              <div
                class="nav-link"
              >
                <a
                  aria-label="link-menu-1"
                  class="sc-gswNZR cEpWoe"
                  href="/"
                >
                  Item 1
                </a>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </div>
</body>
`;
